File: Java/src/main/java/QuestionBank/Solution2201_2300/Solution2239.java
package QuestionBank.Solution2201_2300;
/**
 * @author windows
 */
public class Solution2239 {

    public int findClosestNumber(int[] nums) {
        int min = Integer.MAX_VALUE;
        for (int i = 0; i < nums.length; i++) {
            if (Math.abs(nums[i] - 0) < Math.abs(min)){
                min = nums[i];
            }else if (Math.abs(nums[i] - 0) == Math.abs(min) && nums[i] > min){
                min = nums[i];
            }
        }
        return min;
    }
    
    
    public static void main(String[] args) {
        Solution2239 solution2239 = new Solution2239();
        System.out.println(solution2239.findClosestNumber(new int[]{-4,-2,1,4,8}));
    }
}

File: C++/src/Solution2201_2300/Solution2239.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution2239.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int findClosestNumber(vector<int>& nums) {
  int minVal = INT_MAX;
  int minMax = INT_MIN;
  for (const auto &item : nums){
    if (abs(item - 0) < minVal){
      minVal = abs(item - 0);
      minMax = item;
    } else if (abs(item - 0) == minVal){
      if (minMax < item){
        minMax = item;
      }
    }
  }
  return minMax;
}
